### PR TITLE
Allow cugraph-nx to run networkx tests for nx versions 3.0, 3.1, and 3.2

### PR DESCRIPTION
**Pull request by eriknw**
This is one step to support #3770 

Note that NetworkX version 3.2 is currently under development.

CC @rlratzel

### CODE REVIEW

**Review by rlratzel — COMMENTED** [review]
LGTM, just had some ideas below.

File: python/cugraph-nx/cugraph_nx/interface.py [hunk 1]
```diff
@@ -51,17 +52,110 @@ def key(testpath):
                 return (testname, frozenset({classname, filename}))
             return (testname, frozenset({filename}))
 
-        string_attribute = "unable to handle string attributes"
+        no_weights = "weighted implementation not currently supported"
```
[the remark this answers is not in the document]
Comment (eriknw) [reply python/cugraph-nx/cugraph_nx/interface.py:62]: Done 👍

File: python/cugraph-nx/cugraph_nx/interface.py [hunk 2]
```diff
@@ -51,17 +52,110 @@ def key(testpath):
...
+        no_multigraph = "multigraphs not currently supported"
```
[the remark this answers is not in the document]
Comment (eriknw) [reply python/cugraph-nx/cugraph_nx/interface.py:63]: Yeah, that sounds nice; we could look into cleaning this up and making it easier for any backend to mark tests as xfail.

This seems pretty low priority for now--we'll see if I/we get frustrated by fiddling with this as we add more algorithms, and maybe we'll get motivated to make it better.

**Review by rlratzel — APPROVED** [review]

**Pull request merged**